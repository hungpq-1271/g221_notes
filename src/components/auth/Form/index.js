import React from 'react'
import styled from 'styled-components'
import FormControl from '@material-ui/core/FormControl'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'

const StyledForm = styled(FormControl)`
  position: relative;
  width: 100%;
  margin: auto;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

const SubmitButton = styled(Button)`
  width: 100%;
`

export default function Form({ submitButtonLabel, extraLinkEl, children, onSubmit }) {
  return (
    <StyledForm>
      {children}
      <Grid container spacing={2}>
          <Grid item xs={8}>
            <Typography>
              {extraLinkEl}
            </Typography>
          </Grid>
          <Grid item xs={4}>
            <SubmitButton
              variant="outlined"
              type="submit"
              onClick={onSubmit}
            >
              {submitButtonLabel}
            </SubmitButton>
          </Grid>
        </Grid>
    </StyledForm>
  )
}
