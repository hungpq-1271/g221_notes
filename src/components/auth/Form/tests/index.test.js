import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Form from '../index'

describe('<Form />', () => {
  it('should render and match the snapshot', () => {
    const { container } = render(<Form />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
