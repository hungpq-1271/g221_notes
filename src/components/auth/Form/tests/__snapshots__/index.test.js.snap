// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render and match the snapshot 1`] = `
.c0 {
  position: relative;
  width: 100%;
  margin: auto;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 {
  width: 100%;
}

<div
  class="MuiFormControl-root c0"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <button
        class="MuiButtonBase-root MuiButton-root c1 MuiButton-outlined"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
